### PR TITLE
add samples and documentation for accessLog programatic configuration

### PR DESCRIPTION
**Pull request by aneveu**

### CODE REVIEW

**Review by violetagg — APPROVED** [review]

File: docs/asciidoc/http-server.adoc [hunk 1]
```diff
@@ -339,10 +339,9 @@ include::{examplesdir}/sni/Application.java[lines=18..47]
 
 == HTTP Access Log
 
-The current logging support provides only the https://en.wikipedia.org/wiki/Common_Log_Format[Common Log Format].
+The `HTTP` access log could be enabled programmatically or by configuration. By default, it is disabled.
```
Comment (Buzzardo) [inline docs/asciidoc/http-server.adoc:342]: Change the line to:

You can access the `HTTP` access either programmatically or by configuration. By default, it is disabled.

File: docs/asciidoc/http-server.adoc [hunk 2]
```diff
@@ -366,6 +365,48 @@ You can use the following configuration (for Logback or similar logging framewor
...
+The following example enables it programmatically:
+====
+[source,java,indent=0]
+.{examplesdir}/accessLog/Application.java
+----
+include::{examplesdir}/accessLog/Application.java[lines=18..32]
+----
+====
+
+Calling this method will take precedence over the system property configuration.
```
Comment (Buzzardo) [inline docs/asciidoc/http-server.adoc:377]: Change "will take" to "takes"

File: docs/asciidoc/http-server.adoc [hunk 3]
```diff
@@ -366,6 +365,48 @@ You can use the following configuration (for Logback or similar logging framewor
...
+
+By default, the logging format is https://en.wikipedia.org/wiki/Common_Log_Format[Common Log Format], but you can
+specify a custom one as parameter like in the following example:
```
Comment (Buzzardo) [inline docs/asciidoc/http-server.adoc:380]: Change the line to:

specify a custom one as a parameter, as in the following example:

File: docs/asciidoc/http-server.adoc [hunk 4]
```diff
@@ -366,6 +365,48 @@ You can use the following configuration (for Logback or similar logging framewor
...
+====
+[source,java,indent=0]
+.{examplesdir}/accessLog/CustomLogAccessFormatApplication.java
+----
+include::{examplesdir}/accessLog/CustomLogAccessFormatApplication.java[lines=18..33]
+----
+====
+
+You can also filter `HTTP` access logs using the `AccessLogFactory#createFilter` method, as in the following example:
```
Comment (Buzzardo) [inline docs/asciidoc/http-server.adoc:389]: Change "using" to "by using"

**Review by Buzzardo — CHANGES_REQUESTED** [review]

File: docs/asciidoc/http-server.adoc [hunk 5]
```diff
@@ -374,10 +374,10 @@ include::{examplesdir}/accessLog/Application.java[lines=18..32]
 ----
 ====
 
-Calling this method will take precedence over the system property configuration.
+Calling this method will takes precedence over the system property configuration.
```
Comment (violetagg) [inline docs/asciidoc/http-server.adoc:377]: Remove `will` from `will takes`

**Review by violetagg — CHANGES_REQUESTED** [review]

**Review by violetagg — APPROVED** [review]

File: docs/asciidoc/http-server.adoc [hunk 6]
```diff
@@ -339,10 +339,9 @@ include::{examplesdir}/sni/Application.java[lines=18..47]
 
 == HTTP Access Log
 
-The current logging support provides only the https://en.wikipedia.org/wiki/Common_Log_Format[Common Log Format].
+You can enable the `HTTP` access either programmatically or by configuration. By default, it is disabled.
```
[the remark this answers is not in the document]
Comment (aneveu) [reply docs/asciidoc/http-server.adoc:342]: thanks!

**Review by simonbasle — APPROVED** [review]

**Pull request merged**